pdns/docker-entrypoint.sh

```diff
@@ -33,14 +36,30 @@ if [ "$MYSQL_NUM_TABLE" -eq 0 ]; then
     $MYSQL_COMMAND -D "$PDNS_gmysql_dbname" < /usr/share/doc/pdns/schema.mysql.sql
 fi
 
-# Configure supermasters if needed
-if [ "${SUPERMASTER_IPS:-}" ]; then
-    $MYSQL_COMMAND -D "$PDNS_gmysql_dbname" -e "TRUNCATE supermasters;"
-    MYSQL_INSERT_SUPERMASTERS=''
-    for i in $SUPERMASTER_IPS; do
-        MYSQL_INSERT_SUPERMASTERS="${MYSQL_INSERT_SUPERMASTERS} INSERT INTO supermasters VALUES('${i}', '$(hostname -f)', 'admin');"
-    done
-    $MYSQL_COMMAND -D "$PDNS_gmysql_dbname" -e "$MYSQL_INSERT_SUPERMASTERS"
+if [ "${PDNS_superslave:-no}" == "yes" ]; then
+    # Configure supermasters if needed
+    if [ "${SUPERMASTER_IPS:-}" ]; then
+        $MYSQL_COMMAND -D "$PDNS_gmysql_dbname" -e "TRUNCATE supermasters;"
+        MYSQL_INSERT_SUPERMASTERS=''
+        if [ "${SUPERMASTER_COUNT:-0}" == "0" ]; then
+            SUPERMASTER_COUNT=10
+        fi
+        i=1; while [ $i -le ${SUPERMASTER_COUNT} ]; do
+            SUPERMASTER_HOST=$(echo ${SUPERMASTER_HOSTS} | cut -d' ' -f${i})
```

Comment:
This is not backward compatible, it needs to be `SUPERMASTER_HOST=$(echo ${SUPERMASTER_HOSTS:-} | cut -d' ' -f${i})`